src/boundaries/edgetracking.jl

```diff
@@ -0,0 +1,304 @@
+* `diverge_thresh = 1e-6`: distance threshold for parallel integration
+* `u1`: first initial state (defaults to first point in first entry of `attractors`)
+* `u2`: second initial state (defaults to first point in second entry of `attractors`)
+* `maxiter = 100`: maximum number of iterations before the algorithm stops
+* `abstol = 0.0`: distance threshold for convergence of the updated edge state
+* `T_transient = 0.0`: transient time before the algorithm starts saving the edge track
+* `tmax = Inf`: maximum integration time of parallel trajectories until re-bisection 
+* `Δt = 0.01`: time step passed to [`step!`](@ref) when evolving the two trajectories
+* `ϵ_mapper = nothing`: `ϵ` parameter in [`AttractorsViaProximity`](@ref)
+* `show_progress = true`: if true, shows progress bar and information while running
+* `kwargs...`: additional keyword arguments to be passed to [`AttractorsViaProximity`](@ref)
+
+## Description
+The edge tracking algorithm is a numerical method to find
+an *edge state* or (possibly chaotic) saddle on the boundary between two basins of
+attraction. Introduced by [Battelino1988](@cite) and further described by
+[Skufca2006](@cite), the
+algorithm has been applied to, e.g., the laminar-turbulent boundary in plane Couette
+flow [Schneider2008](@cite), Wada basins [Wagemakers2020](@cite), as well as Melancholia
+states in conceptual [Mehling2023](@cite) and intermediate-complexity [Lucarini2017](@cite) 
+climate models. 
+Relying only on forward integration of the system, it works even in
+high-dimensional systems with complicated fractal basin boundary structures.
+
+The algorithm consists of two main steps: bisection and tracking. First, it iteratively 
+bisects along a straight line in state space between the intial states `u1` and `u2` to find
+the separating basin boundary. The bisection stops when the two updated states are less than
+`bisect_thresh` (Euclidean distance in state space) apart from each other.
+Next, a `ParallelDynamicalSystem` is initialized
+from these two updated states and integrated forward until the two trajectories diverge
+from each other by more than `diverge_thresh` (Euclidean distance). The two final states of
+the parallel integration are then used as new states `u1` and `u2` for a new bisection, and 
+so on, until a stopping criterion is fulfilled. 
+
+Two stopping criteria are implemented via the keyword arguments `maxiter` and `abstol`.
+Either the algorithm stops when the number of iterations reaches `maxiter`, or when the
+state space position of the updated edge point changes by less than `abstol` (in
+Euclidean distance) compared to the previous iteration. Convergence below `abstol` happens
+after sufficient iterations if the edge state is a saddle point. However, the edge state
+may also be an unstable limit cycle or a chaotic saddle. In these cases, the algorithm will
+never actually converge to a point but (after a burn-in) continue populating the set
```

Comment:
What does "burn-in" mean? I have not heard this expression before.

Reply:
To me "burn-in" is a common term to refer to the transient period after initialization of a simulation that is thrown away if one is interested in some kind of converged solution or steady-state. I will change it to "transient period" to be more explicit, ok?

Reply:
yes